crates/transaction-emitter-lib/src/transaction_generator/transaction_mix_generator.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+use crate::transaction_generator::{TransactionGenerator, TransactionGeneratorCreator};
+use aptos_sdk::types::{transaction::SignedTransaction, LocalAccount};
+use rand::prelude::StdRng;
+use rand::Rng;
+use rand_core::{OsRng, SeedableRng};
+
+pub struct TxnMixGenerator {
+    rng: StdRng,
+    txn_mix: Vec<(Box<dyn TransactionGenerator>, usize)>,
+    total_weight: usize,
+}
+
+impl TxnMixGenerator {
+    pub fn new(rng: StdRng, txn_mix: Vec<(Box<dyn TransactionGenerator>, usize)>) -> Self {
+        let total_weight = txn_mix.iter().map(|(_, weight)| weight).sum();
+        Self {
+            rng,
+            txn_mix,
+            total_weight,
+        }
+    }
+}
+
+impl TransactionGenerator for TxnMixGenerator {
+    fn generate_transactions(
+        &mut self,
+        accounts: Vec<&mut LocalAccount>,
+        transactions_per_account: usize,
+    ) -> Vec<SignedTransaction> {
+        let mut picked = self.rng.gen_range(0, self.total_weight);
+        for (gen, weight) in &mut self.txn_mix {
+            if picked < *weight {
```

Comment:
The logic looks weird to me. Let's say we have two types of transaction with weight 50 each. In that case, the total weight is 100. If the picked weight is 30, then we won't generate any transaction in this case but the behavior should be we we generate one type of transaction with equal probability?

Reply:
Ah, I missed the return statement below - SGTM.